TT-Triage - Summarize running operations across cores

miacim opened the pull request:
### Problem description
Hard to figure out which operation was causing hang if there is multiple kernels active

### What's changed
Added new tt-triage script that finds all unique host assigned ids and print them with information about running operation.

miacim committed:
Add dump_running_operations script to summarize running operations across cores

miacim committed:
Remove unnecessary print statement for device unique ID in dump_running_operations script

tt-vjovanovic commented on tools/triage/dump_running_operations.py:
```diff
@@ -0,0 +1,177 @@
+
+def _format_operation(kernel_name: str | None, watcher_kernel_id: int | None) -> str | None:
+    if kernel_name:
+        return kernel_name
```
If we will change to have TTNN operation name in the future, consider formatting this as:
`f"Kernel: {kernel_name}"`

tt-vjovanovic commented on tools/triage/dump_running_operations.py:
```diff
@@ -0,0 +1,177 @@
+                block = device.get_block(location)
+                for risc_name in block.risc_names:
+                    try:
+                        dispatcher_core_data = dispatcher_data.get_core_data(location, risc_name)
```
Depending on "PR race", we should update this to use `dispatcher_data.get_cached_core_data()` for performance improvement :)

miacim committed:
Refactor dump_running_operations script to improve device and core display logic

- Removed unused fields for device and core counts.
- Introduced MAX_DEVICES_DISPLAYED to limit the number of devices shown in summaries.
- Enhanced core location formatting to handle None values gracefully.
- Streamlined data collection from dispatcher, improving clarity and efficiency.

miacim commented on tools/triage/dump_running_operations.py:
```diff
@@ -0,0 +1,177 @@
+def _format_core_location(device_label: str, location: OnChipCoordinate) -> str:
+    user_str = location.to_user_str()
+    location_token = user_str.split()[0] if user_str else user_str
+    return f"{device_label}:{location_token}" if location_token else device_label
```
removed

miacim committed:
Merge branch 'main' into miacim/host_assigned_id
